### DISCUSSION TITLE
Interpreting OR information

### DISCUSSION THREAD

**Discussion by AdamLeaghj** [ORTools4MTEX/ORTools#23] [Q&A] [answered]
Good evening everyone,  
I got the refined OR information as below. I have 3 questions regarding this information.

- What does it mean when my "Closest parent direction" return to be [0 0 0]?
- Instead of relative deviation angle and rotation axis relative to the variant 1, how to retrieve the full information of all 24 variants (just like the information from Morito et al., i.e., parent austenite plane // child martensite plane & parent austenite direction // child martensite direction)
- Why the first variant return with `- 1: 0.00º / [0.3489,-0.0000,-0.0000]`? I think it should be `- 1: 0.00º / [0.0000, 0.00000. 0000]`

Thanks for your help in advance!

```
 -> Detailed information on the selected OR:
    - OR misorientation angle = 43.8465º
 -> Parallel planes
    - Closest parent plane = (3,-1,1)
    - Closest child plane = (2,1,1)
    - Disor. of parallel plane relationship from OR = 0.1214º
 -> Parallel directions
    - Closest parent direction = [0,0,0]
    - Closest child direction = [-1,1,1]
    - Disor. of parallel directions relationship from OR = 0.71184º
 -> OR misorientation rotation axes
    - Parent rot. axis = [-0.0034,0.0513,0.2684]
    - Child rot. axis = [-0.0044,0.0655,0.3427]
    - Disor. of parallel rot. axes relationship from OR = 0º
 -> Angle & rot. axes of unique variants
    - 1: 0.00º / [0.3489,-0.0000,-0.0000]
    - 2: 60.42º / [-0.1792,0.1836,0.2365]
    - 3: 54.67º / [-0.1896,0.1480,0.2528]
    - 4: 8.05º / [-0.1414,0.0000,0.3190]
    - 5: 54.67º / [-0.1480,0.1896,0.2528]
    - 6: 57.31º / [-0.2441,0.0509,0.2441]
    - 7: 58.71º / [-0.2116,0.1793,0.2116]
    - 8: 6.36º / [-0.1901,0.1901,0.2225]
    - 9: 52.72º / [-0.0743,0.2292,0.2523]
    - 10: 53.22º / [-0.1456,0.2218,0.2266]
    - 11: 11.38º / [-0.1170,0.0173,0.3283]
    - 12: 55.68º / [-0.2248,0.0489,0.2623]
    - 13: 11.38º / [-0.0173,0.1170,0.3283]
    - 14: 53.22º / [-0.2218,0.1456,0.2266]
    - 15: 59.57º / [-0.2253,0.0917,0.2501]
    - 16: 15.70º / [-0.2410,0.0750,0.2410]
    - 17: 48.45º / [-0.2358,0.1028,0.2358]
    - 18: 56.85º / [-0.1180,0.2169,0.2465]
    - 19: 52.72º / [-0.2292,0.0743,0.2523]
    - 20: 55.68º / [-0.0489,0.2248,0.2623]
    - 21: 14.90º / [-0.0976,0.0000,0.3350]
    - 22: 56.85º / [-0.2169,0.1180,0.2465]
    - 23: 59.57º / [-0.0917,0.2253,0.2501]
    - 24: 16.10º / [-0.1496,-0.0000,0.3152]
```

**Comment by frankNiessen** [comment, marked as answer]
Hi Adam, 

I think the [0,0,0] direction is a bug in the code. Could you help me out by saving your OR the following way for me to debug?

```
OR = job.p2c;
save("OR.mat","OR");
```
So the angle-axis relationship to the first variant is consistent in that it does not rely on a convention - if you change the definition of the first variant you also change the definition of the remaining variants accordingly. The Morito convention is a convention as the name says, meaning that we need to sort the variants manually. 

What you can do to get the nearest parallel plane normals and directions (the OR deviates from this in real life), is using the command:

`round2Miller(job.p2c.variants)`

This returns:
 
```
    plane parallel   direction parallel    fit
  (111) || (011)     [1-10] || [100]       2.198°
  (111) || (0-1-1)   [01-1] || [100]       2.198°
  (111) || (011)     [10-1] || [-100]      2.198°
  (111) || (0-1-1)   [1-10] || [100]       2.198°
  (111) || (011)     [01-1] || [100]       2.198°
  (111) || (0-1-1)   [10-1] || [-100]      2.198°
 (1-11) || (011)      [011] || [100]       2.198°
 (1-11) || (0-1-1)    [110] || [-100]      2.198°
 (1-11) || (011)     [10-1] || [100]       2.198°
 (1-11) || (0-1-1)    [011] || [100]       2.198°
 (1-11) || (011)      [110] || [-100]      2.198°
 (1-11) || (0-1-1)   [10-1] || [100]       2.198°
(1-1-1) || (0-1-1)    [110] || [100]       2.198°
(1-1-1) || (011)      [101] || [-100]      2.198°
(1-1-1) || (0-1-1)   [01-1] || [-100]      2.198°
(1-1-1) || (011)      [110] || [100]       2.198°
(1-1-1) || (0-1-1)    [101] || [-100]      2.198°
(1-1-1) || (011)     [01-1] || [-100]      2.198°
 (11-1) || (011)      [101] || [100]       2.198°
 (11-1) || (0-1-1)    [011] || [-100]      2.198°
 (11-1) || (011)     [1-10] || [-100]      2.198°
 (11-1) || (0-1-1)    [101] || [100]       2.198°
 (11-1) || (011)      [011] || [-100]      2.198°
 (11-1) || (0-1-1)   [1-10] || [-100]      2.198°
```

You will notice that the parallel planes are neatly sorted and actually match with the Morito convention 

<div class="csl-entry">Morito, S., Tanaka, H., Konishi, R., Furuhara, T., &#38; Maki, T. (2003). The morphology and crystallography of lath martensite in Fe-C alloys. <i>Acta Materialia</i>, <i>51</i>, 1789–1799. https://doi.org/10.1016/j.actamat.2006.07.009</div>


This is since we apply this sorting in MTEX by default when we register a cubic-to-cubic phase transformation. You will also recognize, that the returned list does not match Table 2 in the above reference exactly:

- The planes are often given as for instance (1-1-1) rather than (-1 1 1), which of course is totally equivalent. 
- In the convention we describe the relationship between <011> and <111> directions, whereas the returned table in MTEX shows <011><100> relationships. Since MTEX again does not know about conventions in metallurgical research, it just chooses the first possible solution it comes across. I will see if I can find some time to implement manual specification of the convention for the round2Miller command.

Lastly, the returned value `1: 0.00º / [0.3489,-0.0000,-0.0000]` is one out of an infinite group of correct answers for variant 1. A rotation by 0° is no rotation and it therefore does not matter which rotation axis you choose, all are equally correct.

Best wishes
Frank

**Reply by AdamLeaghj** [reply]
Hi Frank, 
Thanks for your detailed answer. Please see my response below. 

1. The OR file was attached [here](https://usaskca1-my.sharepoint.com/:u:/g/personal/ncz473_usask_ca/EbMxwg3GoTxNhhT60vlZ7pwBuNFOPYYugz0VL8BJnU84SQ?e=xSmUtb)
2. I noticed you mentioned 2 a bit contradicted sentences here: `The Morito convention is a convention as the name says, meaning that we need to sort the variants manually.` and `You will notice that the parallel planes are neatly sorted and actually match with the Morito convention`. I don't know if it's contradicted each other or I misunderstood what you meant. In fact, the resulting table does not return with the same convention as Morito et al. (you can see your table sorted by the order of the closed-packed planes of austenite, whereas Morito's returns with the order of packet convention). Could you please explain this point a bit further?
3. What does column `fit` means in your table?
4. `Since MTEX again does not know about conventions in metallurgical research, it just chooses the first possible solution it comes across.` I'm sorry that I didn't get your point. The <011>//<100> and <011>//<111> are two totally different relationship. Could you come up with a workaround so I can fix it by myself? (My relationship is actually <011> // <759>).

Thanks Frank so much again, I've been struggling with this for a couple days. You truly made my day.

**Reply by frankNiessen** [reply]
Hi again, 

1. I just loaded your OR and executed `ORinfo(OR)` which yielded the following output:
```
 -> Detailed information on the selected OR:
    - OR misorientation angle = 43.8465º
 -> Parallel planes
    - Closest parent plane = (3,-1,1)
    - Closest child plane = (2,1,1)
    - Disor. of parallel plane relationship from OR = 0.1214º
 -> Parallel directions
    - Closest parent direction = [1,2,-1]
    - Closest child direction = [-1,3,-1]
    - Disor. of parallel directions relationship from OR = 0.86725º
 -> OR misorientation rotation axes
    - Parent rot. axis = [-0.0034,0.0513,0.2684]
    - Child rot. axis = [-0.0044,0.0655,0.3427]
    - Disor. of parallel rot. axes relationship from OR = 0º
 -> Angle & rot. axes of unique variants
    - 1: 0.00º / [0.3489,-0.0000,-0.0000]
    - 2: 60.42º / [-0.1792,0.1836,0.2365]
    - 3: 54.67º / [-0.1896,0.1480,0.2528]
    - 4: 8.05º / [-0.1414,0.0000,0.3190]
    - 5: 54.67º / [-0.1480,0.1896,0.2528]
    - 6: 57.31º / [-0.2441,0.0509,0.2441]
    - 7: 58.71º / [-0.2116,0.1793,0.2116]
    - 8: 6.36º / [-0.1901,0.1901,0.2225]
    - 9: 52.72º / [-0.0743,0.2292,0.2523]
    - 10: 53.22º / [-0.1456,0.2218,0.2266]
    - 11: 11.38º / [-0.1170,0.0173,0.3283]
    - 12: 55.68º / [-0.2248,0.0489,0.2623]
    - 13: 11.38º / [-0.0173,0.1170,0.3283]
    - 14: 53.22º / [-0.2218,0.1456,0.2266]
    - 15: 59.57º / [-0.2253,0.0917,0.2501]
    - 16: 15.70º / [-0.2410,0.0750,0.2410]
    - 17: 48.45º / [-0.2358,0.1028,0.2358]
    - 18: 56.85º / [-0.1180,0.2169,0.2465]
    - 19: 52.72º / [-0.2292,0.0743,0.2523]
    - 20: 55.68º / [-0.0489,0.2248,0.2623]
    - 21: 14.90º / [-0.0976,0.0000,0.3350]
    - 22: 56.85º / [-0.2169,0.1180,0.2465]
    - 23: 59.57º / [-0.0917,0.2253,0.2501]
    - 24: 16.10º / [-0.1496,-0.0000,0.3152]
```
I cannot reproduce your error
2. I say that the Morito convention is a convention from metallurgical research, i.e. Morito once sorted the variants according to packets and closed-packed directions and other researchers adapted this convention. MTEX can per se not know about these conventions, however, in the case of cubic-to-cubic phase transformations as in your case, we have build in a default sorting of the variants according to this Morito covention to make life easier for everybody. Is this clearer?

Morito also sorts the variants according to the four austenite planes, which are the foundations for the 4 packets. Try executing 'ORinfo(orientation.KurdjumovSachs(OR.CS,OR.SS));' and you will receive a list of variants that shows the exact same angles as in Table 2 of Morito et al.  Furthermore, when executing 
`
round2Miller(orientation.KurdjumovSachs(OR.CS,OR.SS).variants)
`

you also get the correct list of Miller direction pairs:
 
   ```
 plane parallel   direction parallel    fit
  (111) || (011)     [10-1] || [11-1]      0°
  (111) || (0-1-1)   [10-1] || [11-1]      0°
  (111) || (011)     [01-1] || [-1-11]     0°
  (111) || (0-1-1)   [01-1] || [-1-11]     0°
  (111) || (011)     [1-10] || [-1-11]     0°
  (111) || (0-1-1)   [1-10] || [-1-11]     0°
 (1-11) || (011)     [10-1] || [-1-11]     0°
 (1-11) || (0-1-1)   [10-1] || [-1-11]     0°
 (1-11) || (011)      [110] || [11-1]      0°
 (1-11) || (0-1-1)    [110] || [11-1]      0°
 (1-11) || (011)      [011] || [-1-11]     0°
 (1-11) || (0-1-1)    [011] || [-1-11]     0°
(1-1-1) || (0-1-1)   [01-1] || [11-1]      0°
(1-1-1) || (011)     [01-1] || [11-1]      0°
(1-1-1) || (0-1-1)    [101] || [11-1]      0°
(1-1-1) || (011)      [101] || [11-1]      0°
(1-1-1) || (0-1-1)    [110] || [-1-11]     0°
(1-1-1) || (011)      [110] || [-1-11]     0°
 (11-1) || (011)     [1-10] || [11-1]      0°
 (11-1) || (0-1-1)   [1-10] || [11-1]      0°
 (11-1) || (011)      [011] || [11-1]      0°
 (11-1) || (0-1-1)    [011] || [11-1]      0°
 (11-1) || (011)      [101] || [-1-11]     0°
 (11-1) || (0-1-1)    [101] || [-1-11]     0°
```
matching exactly with the Morito convention. So the variants are correctly ordered.

3. Fit means how well your actual misorientation matches the misorientation that is described in the table with rational orientation relationships (i.e.  (11-1) || (011)     [1-10] || [11-1] ). You can notice in the above table of the K-S relationship, that the fit is 0° as the K-S relationship is a rational orientation relationship. The OR that you provided me with is irrational and about  2.2 degrees away from  (111) || (011), [1-10] || [100] which is the Nishiyama Wassermann (N-W) OR.. In real life the OR is never exactly K-S or N-W, but deviates from that.

4. <011>//<100> and <011>//<111> is not exactly the same, but very close. If you look at the Bain lattice correspondence below, I have marked the first in red and the second in green, you can see that they both roughly match.

![image](https://user-images.githubusercontent.com/41304940/205364457-3181cf35-2fae-4965-951a-fb5015032642.png)

You cannot expect that the ORInfo function will give you exactly the <011> // <759>, as the latter is a quite low symmetry direction and the purpose of ORInfo is to show the deviation from high symmetry planes and directions. What you can try is playing with the penalty parameter.

`round2Miller(OR,'penalty',0.01)`

The smaller you make this, the higher order Miller indices are being favored.

Best wishes
Frank

**Reply by AdamLeaghj** [reply]
Thanks Frank! I'm crystal clear on all the points but the first one turns out very weird result. Same OR.mat file but returned with a different result (and ONLY directions information is wrong!). Have no idea if this affects my final result or not.

![image](https://user-images.githubusercontent.com/118141541/205387397-193d8a52-9fff-443f-a33a-f22e16804e3a.png)

**Reply by AdamLeaghj** [reply]
Can you try again with my data [here](https://usaskca1-my.sharepoint.com/:u:/g/personal/ncz473_usask_ca/EcjtggZG2E1DpzCkzONmrpIBaXJkh54Kc_Dpg1xyRxGM4Q?e=cebJ59)? 

Thanks.

**Reply by AdamLeaghj** [reply]
Hi @frankNiessen,
Could you please take a look on this one? Thanks.

**Reply by frankNiessen** [reply]
Hi Adam, 

I just ran with your data using the commands

```
job = setParentGrainReconstructor(ebsd,grains,Ini.cifPath);
job.p2c = orientation.KurdjumovSachs(job.csParent, job.csChild);
job.calcParent2Child;
```

I obtained this OR:

```
  Bunge Euler angles in degree
     phi1     Phi    phi2
  336.332 7.49753 68.1622
```

and the output of ORinfo is:
```
 -> Detailed information on the selected OR:
    - OR misorientation angle = 45.09º
 -> Parallel planes
    - Closest parent plane = (1,-2,2)
    - Closest child plane = (2,-1,2)
    - Disor. of parallel plane relationship from OR = 0.68301º
 -> Parallel directions
    - Closest parent direction = [2,0,-1]
    - Closest child direction = [2,2,-1]
    - Disor. of parallel directions relationship from OR = 0.62512º
 -> OR misorientation rotation axes
    - Parent rot. axis = [-0.0325,0.0336,0.2700]
    - Child rot. axis = [-0.0413,0.0427,0.3433]
    - Disor. of parallel rot. axes relationship from OR = 0º
 -> Angle & rot. axes of unique variants
    - 1: 0.00º / [0.0000,-0.0000,0.3484]
    - 2: 60.29º / [-0.1730,0.2011,0.2259]
    - 3: 58.50º / [-0.1923,0.1484,0.2498]
    - 4: 6.04º / [-0.0787,-0.0000,0.3394]
    - 5: 58.50º / [-0.1484,0.1923,0.2498]
    - 6: 55.15º / [-0.2445,0.0425,0.2445]
    - 7: 55.46º / [-0.2149,0.1703,0.2149]
    - 8: 5.60º / [-0.2446,0.0418,0.2446]
    - 9: 53.58º / [-0.2357,0.0772,0.2447]
    - 10: 53.70º / [-0.1428,0.2077,0.2406]
    - 11: 10.60º / [-0.1266,0.0161,0.3242]
    - 12: 58.72º / [-0.2275,0.0625,0.2564]
    - 13: 10.60º / [-0.0161,0.1266,0.3242]
    - 14: 53.70º / [-0.2077,0.1428,0.2406]
    - 15: 59.06º / [-0.0910,0.2358,0.2399]
    - 16: 13.98º / [-0.2452,0.0335,0.2452]
    - 17: 52.40º / [-0.2331,0.1126,0.2331]
    - 18: 54.24º / [-0.1085,0.2216,0.2460]
    - 19: 53.58º / [-0.0772,0.2357,0.2447]
    - 20: 58.72º / [-0.0625,0.2275,0.2564]
    - 21: 13.80º / [-0.0164,-0.0000,0.3480]
    - 22: 54.24º / [-0.2216,0.1085,0.2460]
    - 23: 59.06º / [-0.2358,0.0910,0.2399]
    - 24: 15.00º / [-0.1399,-0.0000,0.3191]
```

This seems fine to me, but it looks different from your output. As long as the OR is the same it won't matter for your work, but of course I am interested in why you see this bug. Can you give me some background information? Which version of ORTools, MTEX and MATLAB do you run? Are you running on a Windows, Mac or Linux system?

BW
Frank

**Reply by AdamLeaghj** [reply]
Hi Frank, 

That's sort of inconvenient since I want to extract some information from ORinfo for my publication. I'm using ORTools 2.0.1 associated with mtex 5.8.2. The MatLab version is 9.10 (R2021a) running on Windows. 

Thanks Frank.

**Reply by frankNiessen** [reply]
Hi Adam, 

I will try to find the problem, but it might be tricky. Can you share the script that you are running? Comparing our information from ORInfo it also seems that we might not get to the same orientation relationship stored in job.p2c either.

**Reply by AdamLeaghj** [reply]
Hi Frank,

I attached the dataset, my code, and OR.mat to this [link](https://drive.google.com/file/d/1O85AolhuPiRyOku3wlte42zKi8dt_31o/view?usp=sharing). Please let me know if it doesn't work. Thanks for the help Frank!

**Comment by AdamLeaghj** [comment]
Hello Frank, I know it's a bit of an asking but how is it going so far? Thanks.

**Reply by frankNiessen** [reply]
Hi Adam, I cannot access your files as of now, please grant me access and I will try to have a look.

**Reply by AdamLeaghj** [reply]
Hi Frank, please try again using this [link](https://drive.google.com/file/d/1O85AolhuPiRyOku3wlte42zKi8dt_31o/view)

Thanks.

**Reply by frankNiessen** [reply]
Thanks Adam, I still get a proper output on my computer when running your script on your data. Could you try running this script on the data on a colleagues computer, checking whether you can reproduce your bug or can find a matching solution with the below?

```
 -> Detailed information on the selected OR:
    - OR misorientation angle = 43.8701º
 -> Parallel planes
    - Closest parent plane = (3,-1,1)
    - Closest child plane = (2,1,1)
    - Disor. of parallel plane relationship from OR = 0.049787º
 -> Parallel directions
    - Closest parent direction = [1,2,-1]
    - Closest child direction = [-1,3,-1]
    - Disor. of parallel directions relationship from OR = 0.70884º
 -> OR misorientation rotation axes
    - Parent rot. axis = [-0.0030,0.0524,0.2681]
    - Child rot. axis = [-0.0038,0.0669,0.3424]
    - Disor. of parallel rot. axes relationship from OR = 0º
```

**Reply by AdamLeaghj** [reply]
Thanks Frank. Yeah, that's really strange that it works on my colleague's PC. May need to reinstall Matlab on my local drive. Anyway, thanks so much for your help, Frank!

**Reply by frankNiessen** [reply]
Sorry I could not find the mistake in your case, I hope you can fix it with a fresh install! All the best wishes and no problem at all.
